pkg/controller/replicaset/replica_set_test.go

```diff
@@ -1369,101 +1008,6 @@ func TestPatchPodFails(t *testing.T) {
 	}
 }
 
-func TestPatchExtraPodsThenDelete(t *testing.T) {
```

Comment:
Do we have any integration test to replace this?

Reply:
@enisoc It is partially covered by deleting pod of `TestReplicaSetBasics` integration test. If they look irrelevant, we should add this test back. I have no idea how to test number of `PATCH` operations in integration test environment. Is it possible?

Reply:
I don't see anything in `TestReplicaSetBasics` that would cover this. It looks like in that integration test, the test is the thing setting the Pod as pending deletion; then you check how RS reacts to this.

What this test does is wait until an RS has all the Pods it wants, and then give it another one that it will want to adopt. The expected behavior is that RS should adopt the Pod, but then realize it has too many. So the RS should choose one of the Pods (possibly the new one; possibly one of the old ones) and delete it, so that the total number of Pods comes back to the desired count.

Reply:
Ya, we should add an integration test for this. Perhaps we can verify whether number of ALL pods within the namespace matches the desired count.